pkg/operator/ceph/object/secret/secret.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2016 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package secret
+
+import (
+	"encoding/json"
+	"strings"
+
+	"emperror.dev/errors"
+	"github.com/ceph/go-ceph/rgw/admin"
+)
+
+const (
+	//#nosec G101 -- This is the name of an annotation key, not a secret
+	updateObjectUserSecretAnnotation = "rook.io/source-of-truth"
+	sourceOfTruthSecret              = "secret"
+)
+
+func ValidateCredentials(accessKey, secretKey string, credentials []admin.UserKeySpec) ([]admin.UserKeySpec, error) {
+	// if len is 0 just update the credentials array with the accessKey and secretKey key
+	if len(credentials) == 0 {
+		credentials = []admin.UserKeySpec{
+			{AccessKey: accessKey,
+				SecretKey: secretKey}}
+		return credentials, nil
+	}
+	// match if the credentials array contains the accessKey and secretKey key
+	for i := range credentials {
+		if credentials[i].AccessKey == accessKey && credentials[0].SecretKey == secretKey {
+			return credentials, nil
+		}
+	}
+
+	return credentials, errors.Errorf("secret keys data is invalid, please update the secret with cvalid format")
+}
+
+func UnmarshalKeys(credentials []byte) ([]admin.UserKeySpec, error) {
```

Comment:
There isn't a marshal equivalent to go back the other direction?

Reply:
didnt get it